rotkehlchen/chain/evm/node_inquirer.py

```diff
@@ -730,6 +735,9 @@ def _call_contract(
             method = getattr(contract.caller(block_identifier=block_identifier), method_name)
             result = method(*arguments or [])
         except (Web3Exception, ValueError) as e:
+            if isinstance(e, InvalidAddress):
```

Comment:
This is wrong type of handling. If you want specific handling for `InvalidAddress` make another except clause for it above and handle it similar to what you did in `_query`